doc/voq/Reliable_TSA.md

```diff
@@ -0,0 +1,75 @@
+    1. The TSA/TSB commands on the Line Card updates the CONFIG_DB on the Line Card. This update is picked up and processed by the bgpcfgd process.
+    2. The TSA/TSB commands on the Supervisor causes the Supervisor to SSH to each Line Card in sequence and issue the same command on the Line Card.
+
+This implementation of the commands on the Supervisor is not adequate for a Chassis System. Some issues -
+   1. There could be a decent amount of delay when the First Line Card acts on the command to when the last Line Card acts on it. A Unresponsive or a slow to respond Line Card can cause significant delays to the Line Cards behinnd it.
+   2. An unresponsive Line Card will never receive and act on the command. This will be the case even if the Line is rebooted.
+   3. Using the TSA/TSB commands have the effect of erasing the TSA/TSB configuration on the Line Cards. This makes it difficult to operate different Line Cards in a chassis with different TSA settings.
+
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+This sections describes the requirements for the TSA on the Supervisor and the Line Cards. These requirements only affect the VOQ Chassis systems. Specifically T0/T1 non Chassis system should not be impaced.
+1. The TSA is an independent configuration attribute (tsa_enabled) on the Supervisor host CONFGI_DB and the per asic CONFIG_DB on the Line Card. 
+2. TSA/TSB commands on the Supervisor should only change the Supervisor host config.
+3. TSA/TSB commands on the Line Card should only change the config of the asics on that line card.
+4. Changes to Supervisor config should also be reflected into the CHASSIS_APP_DB. The asic instances of all Line Cards subscribe to the tsa_enabled attribute of the CHASSIS_APP_DB.
+5. The net operational TSA state of BGP on the asics of a LC is a function of the TSA configuration of the Supervisor and the LC
+6. If Supervisor tsa_enabled == TRUE, operational state is TSA(TRUE). In this configuration -
+   A. After the Line card reboots - the operational state will be TSA when the Line card comes back up
+   B. After "Config reload" - the operational state will be TSA.
+   C. After any of the dockers get restarted (crash, service restart etc..) - the operational state will be TSA.
+7. If Supervisor tsa_enabled == FALSE, operational TSA state is controlled by LC tsa_enabled config
+8. If Supervisor config changes from tsa_enabled == TRUE to tsa_enabled == FALSE : Operational state should be changed only if LC tsa_enabled == FALSE
+9. Should support saving Supervisor TSA configuration host config_db.json file (on the Supervisor)
+10. Supervisor reboot or config reload should restore state from the host config_db.json file of the Supervisor.
+
+# 2 Design Details
+The following changes are planned.
+1. Add a new tsa_enabled attribute to CHASSIS_APP_DB
```

Comment:
new config_db attribute also to be defined to store the user configured tsa state ?

Reply:
It will be the currently defined tsa_enabled attribute. It is currently only set in the LC CONFIG_DB. The Supervisor attribute will be updated as part of the changes for this feature